src/errors/InvalidResourceError.ts

```diff
@@ -5,10 +5,16 @@
 
 // eslint-disable-next-line import/prefer-default-export
 export class InvalidResourceError extends Error {
+    readonly isInvalidResourceError: boolean;
+
     constructor(message = 'Invalid Resource') {
         // Node Error class requires passing a string message to the parent class
         super(message);
         Object.setPrototypeOf(this, InvalidResourceError.prototype);
+        this.isInvalidResourceError = true;
         this.name = this.constructor.name;
     }
 }
+export function isInvalidResourceError(err: InvalidResourceError) {
```

Comment:
These kind of functions should be type guards

https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards